`Arg(T, 'name')` not equivalent to `T` in `Callable`

Given the code

```python
T = TypeVar("T", bound="Foo")
P = ParamSpec("P")

# this class serves as a decorator
class print_y_first:
    def __call__(
        self, method: Callable[Concatenate[T, P], str]
    ) -> Callable[Concatenate[T, P], str]:
        def wrapper(_self: T, *args: P.args, **kwargs: P.kwargs) -> str:
            print(_self.y)
            return method(*args, **kwargs)

        return wrapper


class Foo:

    y: int

    @print_y_first()
    def stringify(self, x: int) -> str:
        return str(x)
```

Mypy reports

```
error: Incompatible return value type (got "Callable[[Arg(T, '_self'), **P], str]", expected "Callable[[T, **P], str]")  [return-value]
note: This may be because "wrapper" has arguments named: "_self"
```

It does not recognize that 

```python
Callable[[Arg(T, '_self'), **P], str] == Callable[[T, **P], str]
```

Mypy version: `mypy 0.971 (compiled: yes)`
Mypy config options: `--strict=True`
Python version: 3.10.4

--- a/mypy/messages.py
+++ b/mypy/messages.py
@@ -791,8 +791,8 @@ class MessageBuilder:
             if names:
                 missing_arguments = '"' + '", "'.join(names) + '"'
                 self.note(
-                    f'This is likely because "{original_caller_type.name}" has arguments '
-                    f"named: {missing_arguments}. Consider marking them positional-only",
+                    f'This is likely because "{original_caller_type.name}" has named arguments: '
+                    f"{missing_arguments}. Consider marking them positional-only",
                     context,
                     code=code,
                 )

--- a/mypy/messages.py
+++ b/mypy/messages.py
@@ -791,8 +791,8 @@ class MessageBuilder:
             if names:
                 missing_arguments = '"' + '", "'.join(names) + '"'
                 self.note(
-                    f'This may be because "{original_caller_type.name}" has arguments '
-                    f"named: {missing_arguments}",
+                    f'This is likely because "{original_caller_type.name}" has arguments '
+                    f"named: {missing_arguments}. Consider marking them positional-only",
                     context,
                     code=code,
                 )

--- a/test-data/unit/check-parameter-specification.test
+++ b/test-data/unit/check-parameter-specification.test
@@ -530,7 +530,7 @@ reveal_type(transform(bar)) # N: Revealed type is "def (builtins.str, *args: bui
 def expects_int_first(x: Callable[Concatenate[int, P], int]) -> None: ...
 
 @expects_int_first  # E: Argument 1 to "expects_int_first" has incompatible type "Callable[[str], int]"; expected "Callable[[int], int]" \
-                    # N: This may be because "one" has arguments named: "x"
+                    # N: This is likely because "one" has arguments named: "x". Consider marking them positional-only
 def one(x: str) -> int: ...
 
 @expects_int_first  # E: Argument 1 to "expects_int_first" has incompatible type "Callable[[NamedArg(int, 'x')], int]"; expected "Callable[[int, NamedArg(int, 'x')], int]"
@@ -598,7 +598,7 @@ f2(lambda x: 42)(42, x=42)
 main:10: error: invalid syntax; you likely need to run mypy using Python 3.8 or newer
 [out version>=3.8]
 main:17: error: Incompatible return value type (got "Callable[[Arg(int, 'x'), **P], R]", expected "Callable[[int, **P], R]")
-main:17: note: This may be because "result" has arguments named: "x"
+main:17: note: This is likely because "result" has arguments named: "x". Consider marking them positional-only
 
 [case testNonStrictParamSpecConcatenateNamedArgs]
 # flags: --python-version 3.8

--- a/test-data/unit/check-parameter-specification.test
+++ b/test-data/unit/check-parameter-specification.test
@@ -530,7 +530,7 @@ reveal_type(transform(bar)) # N: Revealed type is "def (builtins.str, *args: bui
 def expects_int_first(x: Callable[Concatenate[int, P], int]) -> None: ...
 
 @expects_int_first  # E: Argument 1 to "expects_int_first" has incompatible type "Callable[[str], int]"; expected "Callable[[int], int]" \
-                    # N: This is likely because "one" has arguments named: "x". Consider marking them positional-only
+                    # N: This is likely because "one" has named arguments: "x". Consider marking them positional-only
 def one(x: str) -> int: ...
 
 @expects_int_first  # E: Argument 1 to "expects_int_first" has incompatible type "Callable[[NamedArg(int, 'x')], int]"; expected "Callable[[int, NamedArg(int, 'x')], int]"
@@ -598,7 +598,7 @@ f2(lambda x: 42)(42, x=42)
 main:10: error: invalid syntax; you likely need to run mypy using Python 3.8 or newer
 [out version>=3.8]
 main:17: error: Incompatible return value type (got "Callable[[Arg(int, 'x'), **P], R]", expected "Callable[[int, **P], R]")
-main:17: note: This is likely because "result" has arguments named: "x". Consider marking them positional-only
+main:17: note: This is likely because "result" has named arguments: "x". Consider marking them positional-only
 
 [case testNonStrictParamSpecConcatenateNamedArgs]
 # flags: --python-version 3.8

